api/queries_repo.go

```diff
@@ -1198,3 +1199,24 @@ func CreateRepoTransformToV4(apiClient *Client, hostname string, method string,
 		IsPrivate: responsev3.Private,
 	}, nil
 }
+
+func RepoTagExists(httpClient *http.Client, repo ghrepo.Interface, tagName string) (bool, error) {
```

Comment:
Naming nitpick: this method looks like it looks up the existence of any fully qualified git ref and is not just limited to tags. Can the name reflect that, e.g. maybe it can be `RepoRefExists`?

Second, since this method is only used from a single package, maybe it can be inlined in the same package itself? This is to avoid having the `api` package grow too much since it's already busy.

Reply:
Yeah I can do that. I had put it in `api` as it has `graphQLClient()` which is just a wrapper to get a `shurcool/graphql` client, but I didn't know how I felt about spreading around the usage of `shurcool/graphql` outside the `api` package.